Example/Messaging/Sample/Sample/ContentView.swift

```diff
@@ -42,8 +43,8 @@ struct ContentView: View {
               .layoutPriority(1)
           }
 
-          NavigationLink(destination: DetailView()) {
-            Text("Show Detail View")
+          NavigationLink(destination: SettingsView(settings: self.settings)) {
```

Comment:
Since `settings` is passed in via the environment, you should be able to ignore it in this struct entirely and just put `@EnvironmentObject var settings: UserSettings` in the `SettingsView` itself.

Reply:
But UserSettings is also used in SessionDelegate so I created the instance and passing in as a second environmentObject in contentView. (first one is identity)

Reply:
Right - adding an `.environmentObject()` on a view adds it for all children of that view.  The `settings` variable here isn't actually used in the `ContentView` struct (we don't access the variable outside of passing it through to `SettingsView`) but since `SettingsView` has `ContentView()` as an ancestor which has `UserSettings` in it's environment, it'll automatically get access to that environment variable as well. The resultant state (concatenated, missing other parts) can be like:

In the `SceneDelegate`:
```
UIHostingController(rootView: contentView
  // Pass in the settings as an environmentObject to `contentView`
  .environmentObject(settings)
)
```

In the `ContentView` struct:
```
// No need to declare the `@EnvironmentObject var settings: UserSettings` in this struct
// since it's not used.

// No need to pass in the `UserSettings` here since it already has a copy in the environment.
NavigationLink(destination: SettingsView()) {
  Text("Settings")
}
```

In the `SettingsView` struct:
```
// Pull the settings from the environment here, no need to inject in the constructor
@EnvironmentObject var settings: UserSettings

// Remove the @ObservedObject since we have access through the environment.
```

Reply:
Nice it works! So @environmentObject is kinda global in this file?

Reply:
Global for all subviews - yes!